### DISCUSSION TITLE
how can I host my page with private repository?

### DISCUSSION THREAD

**Discussion by mshaheerz** [community/community#27777] [Other Feature Feedback, Questions, & Ideas] [answered]
I cant see Page option in My private repository

**Comment by melvincwng** [comment]
@mshaheerz, follow these steps to host your private repository on GitHub pages:

1. Go to your code repository on GitHub
2. Click 'Settings'
3. On the left sidebar, find the tab 'Pages' (You will end up on GitHub Pages section)
4. Build and deploy your application from the relevant 'Source' and 'Branch'
5. Wait around 10-15mins for your app to be build and deployed. Once it's done, refresh the page to find the URL where your web app is deployed.

If these steps have been useful, kindly appreciate it if you could 'Mark as answer' so that others know this issue has been addressed.

Thanks, and have a nice day ^^

**Comment by tanukrishna4** [comment, marked as answer]
I contacted GitHub support and they replied back confirming this. They stated "You can host a GitHub Pages site from a private repository, however all published sites are public." It's only possible though when you have the Pro account

**Reply by EliasAfara** [reply]
If you are not interested in upgrading to pro, you could checkout Netlify. I recommend watching [this](https://www.youtube.com/watch?v=KIvDWeTXbQk&ab_channel=JoshuaMorony) youtube video which showcases how to set up a CI/CD Pipeline with GitHub Actions & Netlify.

**Comment by jfullstackdev** [comment]
yes, it's only possible if you are not in the free account
